rs/sns/governance/src/governance.rs

```diff
@@ -2007,6 +2017,95 @@ impl Governance {
             .unwrap_or(u64::MAX);
     }
 
+    pub async fn get_sns_statistics(
+        &self,
+        request: GetSnsStatusRequest,
+    ) -> Result<GetSnsStatusResponse, GovernanceError> {
+        let num_recent_proposals = self.recent_proposals(request.time_window_seconds.unwrap());
+        let last_transaction_timestamp = self.get_most_recent_tx_ts().await?;
+
+        // transaction timestamps are in nanoseconds
+        Ok(GetSnsStatusResponse {
+            num_recent_proposals: Some(num_recent_proposals),
+            last_transaction_timestamp: Some(last_transaction_timestamp),
+        })
+    }
+
+    async fn get_most_recent_tx_ts(&self) -> Result<u64, GovernanceError> {
```

Comment:
For the low-level function, let's have the following signature:

```suggestion
    async fn get_latest_block_timestamp_seconds(&self) -> Result<u64, String> {
```

The caller of this function could then map the potential error message into a `GovernanceError`.

Reply:
Done.